### PR TITLE
Added Clear Button to History and WebSockets Tab

### LINKED ISSUE #411: Add option to clear history / websockets tab
This would allow users to ignore older messages they are no longer interested in.

### PR DESCRIPTION
**Pull request by Vitika9**
Fixes #411 
Signed-off-by: Vitika9 <vitika.program@gmail.com>

### CODE REVIEW

**Comment by thc202** [comment]
The changelog needs to be updated. I didn't try the changes yet.

**Comment by psiinon** [comment]
I'm having problems getting the HUD to run correctly, which is not a problem with this PR.
Can you post a screenshot of how the button appears?
I'm not seeing it yet, but that could be a local problem...

**Comment by psiinon** [comment]
I can run the HUD via the weekly, so must be a build issue for me.
However when I copy your changes in I dont see the new button anywhere :/
Screenshot 2023-01-18 at 11-33-52 The BodgeIt Store
What am I missing?

**Comment by Vitika9** [comment]
> However when I copy your changes in I dont see the new button anywhere :/ 

The new button is at the top right, I don't know why it isn't visible at your end :(
image

**Comment by psiinon** [comment]
Thanks! I'll have another go...

**Comment by psiinon** [comment]
conflicting now :/

**Comment by Vitika9** [comment]
> conflicting now :/

Done!

**Comment by Vitika9** [comment]
> It seems that clicking the clear button in the History tab also clears the WebSockets tab (and vice versa), is that the expected behaviour? I thought they would work independently (or I'm doing something wrong?). Using `ws.ifelse.io/.ws` for tests.

My bad, I was using the same event for both of them. Fixed now 😀

**Comment by Vitika9** [comment]
@psiinon @thc202 do I need to make more changes? Please review.

File: src/main/zapHomeFiles/hud/drawer.html [hunk 1]
```diff
@@ -54,6 +54,11 @@
                         <th class="title hidden">
                             <span>{{ historyItemsFilteredMessage }}</span>
                         </th>
+                        <th class="title">
+                            <span>
+                                <button @click="clearHttpMessages()">Clear</button>
```
Comment (psiinon) [inline src/main/zapHomeFiles/hud/drawer.html:59]: The GUI should not have hard coded display strings in it (like "Clear") - those cannot be i18ned. Instead use something like `:name="$t('message.history_tool')"` - thse tags are defined in src/other/resources/UIMessages/UIMessages.properties

**Review by psiinon — CHANGES_REQUESTED** [review]

File: src/main/zapHomeFiles/hud/drawer.html (hunk 1 above)
Comment (psiinon) [reply src/main/zapHomeFiles/hud/drawer.html:59]: The rest of the HTML uses these so you should be able to find plenty of examples, but if not just let us know.
Othert than that the PR seems to work well :)

Comment (Vitika9) [reply src/main/zapHomeFiles/hud/drawer.html]: Done, thanks 😀

**Review by psiinon — APPROVED** [review]
LGTM!

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by psiinon** [comment]
@Vitika9 FYI we have just released the HUD with this change - I've installed it from the market place and it works well!
Many thanks - really good to see the HUD progressing again.
What are your details for the ZAP credits? 
https://www.zaproxy.org/docs/desktop/credits/#zap-extended-team

**Comment by Vitika9** [comment]
@psiinon I am glad to see my changes reflect there, will contribute more! 😃
Here are my details: Vitika Soni, https://github.com/Vitika9

**Pull request merged**